src/collection.js

```diff
@@ -610,6 +610,23 @@ export default class Collection {
     );
   }
 
+  /**
+   * Soft Delete all records from the local database.
+   *
+   * @return {Promise}
+   */
+  async deleteAll() {
+    const records = await this.db.list();
+    // await results.forEach( (record) => {
+    //     delete(record.id);
+    // });
+    // this._queueEvent("deleteAll", { data: [] });
+    // return { data: [], permissions: {} };
+    return this.execute(transaction => {
+      return transaction.deleteAll(records);
+    }, {});
```

Comment:
I believe we can list and delete in the same transaction. In other words, I think the transaction.deleteAll() could be without parameters maybe

Reply:
I didn't find examples of that, and I thought it might be useful to parameterize those records, maybe it can be utilized by a future functionality.